ts/src/main/scala/ch/epfl/ts/evaluation/Evaluator.scala

```diff
@@ -0,0 +1,152 @@
+package ch.epfl.ts.evaluation
+
+import scala.language.postfixOps
+import scala.collection.mutable.{Map => MMap, MutableList => MList}
+import scala.concurrent.duration.{DurationInt, DurationLong}
+import scala.concurrent.duration.FiniteDuration
+import akka.actor.{ActorRef, Cancellable}
+import ch.epfl.ts.component.{ComponentRegistration, Component, ComponentRef}
+import ch.epfl.ts.data._
+import ch.epfl.ts.data.Currency._
+
+/**
+  * Evaluates the performance of traders by total returns, volatility, draw down and sharpe ratio
+  *
+  * To use this class, redirect all previous connections into and out of the trader to
+  * instances of this class.
+  *
+  * Note that a component providing Quote data must be connected to this component in order for
+  * it to maintain a price table and calculate profit/loss correctly.
+  *
+  * @param trader the reference to the trader component
+  * @param traderId the id of the trader
+  * @param initial the initial seed money
+  * @param currency currency of the initial seed money
+  * @param period the time period to compute performance
+  */
+class Evaluator(trader: ComponentRef, traderId: Long, initial: Double, currency: Currency, period: FiniteDuration) extends Component {
+  // for usage of scheduler
+  import context._
+
+  private var schedule: Cancellable = null
+  private val wallet = MMap[Currency, Double](currency -> initial)
+  private val returnsList = MList[Double]()
+  private val priceTable = MMap[(Currency, Currency), Double]()
+
+  private var lastValue = initial
+  private var maxProfit = 0.0
+  private var maxLoss = 0.0
+
+  //TODO find appropriate value for risk free rate
+  val riskFreeRate = 0.03
+  def totalReturns: Double = value() / initial
+  def volatility: Double = computeVolatility
+  def drawdown: Double = maxLoss / initial
+  def sharpeRatio: Double = (totalReturns - riskFreeRate) / volatility
+
+  /**
+   * Redirects out-going connections to the trader
+   * */
+  override def connect(ar: ActorRef, ct: Class[_], name: String) = {
+    trader.ar ! ComponentRegistration(ar, ct, name)
+  }
+
+  /**
+   * Handles interested messages and forward all messages to the trader
+   */
+  override def receiver = {
+    case t: Transaction if t.buyerId == traderId =>  // buy
+      trader.ar ! t
+      buy(t)
+    case t: Transaction if t.sellerId == traderId =>  // sell
+      trader.ar ! t
+      sell(t)
+    case q: Quote =>
+      updatePrice(q)
+    case 'UpdateStatistics =>
+      updateStatistic
+    case m => trader.ar ! m
+  }
+
+  /**
+   *  Returns the exchange ratio between two currency
+   * */
+  private def ratio(from: Currency, to: Currency): Double = {
+    if (from == to) 1.0 else priceTable(from -> to)
+  }
+
+  /**
+   *  Returns the total money of the wallet converted to the given currency
+   * */
+  private def value(in: Currency = currency): Double = {
+    (wallet :\ 0.0) { case ((c, amount), acc) => acc + ratio(c, in) * amount }
```

Comment:
Actually could you expand this to the full syntax? `:\` is pretty but I'm afraid it would be confusing for people trying to take over our codebase.